segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -1769,7 +1770,7 @@ private void write00(SystemSnapshotRecord object, RevisionDataOutput output) thr
             private void read00(RevisionDataInput input, SystemSnapshotRecord.SystemSnapshotRecordBuilder b) throws IOException {
                 b.epoch(input.readCompactLong());
                 b.fileIndex(input.readCompactInt());
-                b.segmentSnapshotRecords(input.readCollection(ELEMENT_DESERIALIZER));
+                b.segmentSnapshotRecords(input.readCollection(ELEMENT_DESERIALIZER, () -> new Vector<>()));
```

Comment:
Why is this needed? Is `segmentSnapshotRecords` modified after it is built?

Reply:
same as above

Reply:
Ok. Great. Then let's use: ImmutableList and use ImmutableList.builder() it will make this explicit and have higher performance